packages/catalog-server/src/lib/graphql.ts

```diff
@@ -0,0 +1,127 @@
+        const packageInfo = await catalog.getPackageInfo(packageName);
+        if (packageInfo === undefined) {
+          console.log(`package ${packageName} not found in db, importing`);
+          let result;
+          try {
+             result = await catalog.importPackage(packageName);
+          } catch (e) {
+            console.error(e);
+            throw e;
+          }
+          const {packageInfo} = result;
+          if (isReadablePackage(packageInfo)) {
+            console.log(`package ${packageName} is readable`);
+            return packageInfo;
+          } else {
+            console.log(`package ${packageName} is not readable`);
+            return null;
+          }
+        } else if (isReadablePackage(packageInfo)) {
+          console.log(`package ${packageName} found in db and readable`);
+          return packageInfo;
+        } else {
+          console.log(`package ${packageName} found in db and not readable`);
+          return null;
+        }
+      },
+    },
+    PackageInfo: {
+      __resolveType(obj) {
+        if (isReadablePackage(obj)) {
+          return 'ReadablePackageInfo';
+        } else {
+          return 'UnreadablePackageInfo';
+        }
+      },
+    },
+    ReadablePackageInfo: {
+      version: async (packageInfo, {versionOrTag}, context, _info) => {
+        // TODO(justinfagnani): strongly type context?
+        context.packageName = packageInfo.name;
+        console.log('PackageInfo version', packageInfo.name, versionOrTag);
+
+        // Check to see if versionOrTag is a distTag
+        const distTags = packageInfo.distTags;
+        const version =
+          distTags?.find((distTag) => distTag.tag === versionOrTag)?.version ??
+          versionOrTag;
+
+        const packageVersion = await catalog.getPackageVersion(packageInfo.name, version);
+        if (packageVersion === undefined) {
+          throw new Error(`tag ${packageInfo.name}@${versionOrTag} not found`);
+        }
+        return packageVersion;
+      },
+    },
+    PackageVersion: {
+      __resolveType(obj) {
+        if (isReadablePackageVersion(obj)) {
+          return 'ReadablePackageVersion';
+        } else {
+          return 'UnreadablePackageVersion';
+        }
+      },
+    },
+    ReadablePackageVersion: {
+      customElements: async (
+        packageVersion,
+        {tagName}: {tagName?: string | null},
+        context,
+        _info
+      ) => {
+        // const packageName = packageVersion.name;
+        const packageName = context.packageName;
+        const version = packageVersion.version;
+        console.log(`Query.ReadablePackageVersion.customElements ${packageName} ${tagName}`);
+        return catalog.getCustomElements(
+          packageName,
+          version,
+          tagName ?? undefined
+        );
+      },
+    },
+    Mutation: {
+      async importPackage(_parent, {packageName}: {packageName: string}) {
+        console.log('mutation deletePackage', packageName);
```

Comment:
Not sure I understand this -- why do we log deletePackage and then just return a package?

Reply:
Oh, this is leftover code from the prototype. I'll remove this.

Reply:
Actually, I think this is a good point to put some explicit import extras, like reporting problems. I'll add a TODO and comment in here.